### PR TITLE
Replace calls to fprintf in the logging and transaction code.

### PR DESCRIPTION
**Pull request by keithbostic**
@ddanderson, would you please review and merge this one?

It's a trivial cleanup, but there's a change in dist/log.py that I thought might be worth a few seconds.

Thanks!

### CODE REVIEW

File: src/btree/bt_debug.c [hunk 1]
```diff
@@ -204,8 +204,8 @@ __wt_debug_addr_print(
 	WT_DECL_ITEM(buf);
 
 	WT_RET(__wt_scr_alloc(session, 128, &buf));
-	fprintf(stderr, "%s\n",
-	    __wt_addr_string(session, addr, addr_size, buf));
+	WT_RET(__wt_fprintf(stderr, "%s\n",
+	    __wt_addr_string(session, addr, addr_size, buf)));
 	__wt_scr_free(session, &buf);
```
Comment (ddanderson) [inline src/btree/bt_debug.c]: @keithbostic, we probably want to make sure the `buf` scratch buffer is freed if there's an error from `__wt_fprintf`.

File: src/log/log_auto.c [hunk 2]
```diff
@@ -131,11 +131,14 @@ __wt_logop_col_put_print(
...
+	WT_RET(__wt_fprintf(out, " \"optype\": \"col_put\",\n"));
+	WT_RET(__wt_fprintf(out,
+	    "        \"fileid\": \"%" PRIu32 "\",\n", fileid));
+	WT_RET(__wt_fprintf(out,
+	    "        \"recno\": \"%" PRIu64 "\",\n", recno));
 	WT_RET(__logrec_jsonify_str(session, &escaped, &value));
-	fprintf(out, "        \"value\": \"%s\"", escaped);
+	WT_RET(__wt_fprintf(out,
+	    "        \"value\": \"%s\"", escaped));
 	__wt_free(session, escaped);
```
Comment (ddanderson) [inline src/log/log_auto.c]: @keithbostic, `escaped` should be freed if there's an error from __wt_fprintf`

File: src/log/log_auto.c [hunk 3]
```diff
@@ -369,10 +380,12 @@ __wt_logop_row_remove_print(
...
-	fprintf(out, " \"optype\": \"row_remove\",\n");
-	fprintf(out, "        \"fileid\": \"%" PRIu32 "\",\n", fileid);
+	WT_RET(__wt_fprintf(out, " \"optype\": \"row_remove\",\n"));
+	WT_RET(__wt_fprintf(out,
+	    "        \"fileid\": \"%" PRIu32 "\",\n", fileid));
 	WT_RET(__logrec_jsonify_str(session, &escaped, &key));
-	fprintf(out, "        \"key\": \"%s\"", escaped);
+	WT_RET(__wt_fprintf(out,
+	    "        \"key\": \"%s\"", escaped));
 	__wt_free(session, escaped);
```
Comment (ddanderson) [inline src/log/log_auto.c]: @keithbostic, likewise there's a few more of these `escaped` that need to be freed on error.

**Comment by ddanderson** [comment]
@keithbostic, looks good, modulo some cleanups on error.

**Comment by keithbostic** [comment]
@ddanderson, thanks, I should have looked more carefully (although, to be kind, we already leaked memory on error in some of these functions).

Anyway, can you please take another look? Unfortunately, I had to change log.py extensively, so it probaby needs real review.

**Comment by ddanderson** [comment]
@keithbostic, that looks great, I'm merging.

**Pull request merged**